### PR TITLE
fix: ledger don't verify manual receive address

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

Two birds, one stone:

- The receive address that's displayed at Ledger verify addresses step is currently wrong for EVM chains i.e not the manual receive address
- The buy asset verification should *not* be triggered altogether, as it is potentially impossible to verify a manual receive address

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

N/A, spotted by @0xdef1cafe, here's your fix ser

> What protocols, transaction types or contract interactions might be affected by this PR?

Low - ensure this still works as expected without manual receive address

## Testing

- When you have set a manual receive address, the buy asset verification step isn't displayed

### Engineering

- ^

### Operations

- ^

### CODE REVIEW

**Comment by gomesalexandre** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/shapeshift/web/6399?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#6399** <a href="https://app.graphite.dev/github/pr/shapeshift/web/6399?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#6398** <a href="https://app.graphite.dev/github/pr/shapeshift/web/6398?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @gomesalexandre and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by kaladinlight — CHANGES_REQUESTED** [review]
image

the logic and view seem inverted. we are still trying to verify the buy asset, but only showing the sell address.

**Comment by gomesalexandre** [comment]
> image
> 
> the logic and view seem inverted. we are still trying to verify the buy asset, but only showing the sell address.

Nice catch! Did some little cleanup while at it which fixed things there.

- No manual receive address

Screenshot 2024-03-12 at 21 17 27


- Manual receive address

 
Screenshot 2024-03-12 at 21 18 36

https://github.com/shapeshift/web/pull/6399/commits/2c36bc19f6ef18feb4588e75383f1bb83a651947 / https://github.com/shapeshift/web/pull/6399/commits/26ecc9d8d7918e0e295a7f5984bcb965e4041893

File: src/components/MultiHopTrade/components/VerifyAddresses/VerifyAddresses.tsx [hunk 1]
```diff
@@ -141,15 +143,6 @@ export const VerifyAddresses = () => {
...
       } else if (type === 'buy') {
-        if (!shouldVerifyBuyAddress) {
-          return (
-            maybeManualReceiveAddress &&
-            setVerifiedAddresses(
-              new Set([...verifiedAddresses, maybeManualReceiveAddress.toLowerCase() ?? '']),
-            )
-          )
-        }
-
```
Comment (gomesalexandre) [inline src/components/MultiHopTrade/components/VerifyAddresses/VerifyAddresses.tsx:152]: Now useless - part of the issue we were seeing

File: src/components/MultiHopTrade/components/VerifyAddresses/VerifyAddresses.tsx [hunk 2]
```diff
@@ -87,15 +90,14 @@ export const VerifyAddresses = () => {
     [isAddressVerified, sellAddress],
   )
   const buyVerified = useMemo(
-    () => isAddressVerified(buyAddress ?? ''),
-    [buyAddress, isAddressVerified],
+    () => maybeManualReceiveAddress || isAddressVerified(buyAddress ?? ''),
```
Comment (kaladinlight) [inline src/components/MultiHopTrade/components/VerifyAddresses/VerifyAddresses.tsx:93]: nitpick refine type (boolean your fav! haha)

```suggestion
    () => Boolean(maybeManualReceiveAddress) || isAddressVerified(buyAddress ?? ''),
```

**Review by kaladinlight — APPROVED** [review]

File: src/components/MultiHopTrade/components/VerifyAddresses/VerifyAddresses.tsx (hunk 2 above)
Comment (gomesalexandre) [reply src/components/MultiHopTrade/components/VerifyAddresses/VerifyAddresses.tsx:93]: I like Boolean

**Pull request merged**